File: Tugas Video/PROJECT ANDROID STUDIO (PLAYLIST 2)/Wave Of Food User Dan Admin/WaveOfFoodAdmin/app/src/main/java/com/kelasxi/waveoffoodadmin/LoginActivity.java
package com.kelasxi.waveoffoodadmin;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class LoginActivity extends AppCompatActivity {
    
    private EditText etEmail, etPassword;
    private Button btnLogin;
    private ProgressBar progressBar;
    private FirebaseAuth auth;
    
    // Admin credentials - dalam implementasi nyata, gunakan Firebase Authentication rules
    private static final String ADMIN_EMAIL = "admin@waveoffood.com";
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        
        // Initialize Firebase Auth
        auth = FirebaseAuth.getInstance();
        
        // Check if user is already logged in
        FirebaseUser currentUser = auth.getCurrentUser();
        if (currentUser != null && isAdminUser(currentUser.getEmail())) {
            navigateToMainActivity();
            return;
        }
        
        initViews();
        setupClickListeners();
    }
    
    private void initViews() {
        etEmail = findViewById(R.id.et_email);
        etPassword = findViewById(R.id.et_password);
        btnLogin = findViewById(R.id.btn_login);
        progressBar = findViewById(R.id.progress_bar);
    }
    
    private void setupClickListeners() {
        btnLogin.setOnClickListener(v -> loginAdmin());
    }
    
    private void loginAdmin() {
        String email = etEmail.getText().toString().trim();
        String password = etPassword.getText().toString().trim();
        
        if (email.isEmpty()) {
            etEmail.setError("Email tidak boleh kosong");
            etEmail.requestFocus();
            return;
        }
        
        if (password.isEmpty()) {
            etPassword.setError("Password tidak boleh kosong");
            etPassword.requestFocus();
            return;
        }
        
        // Check if email is admin email
        if (!isAdminUser(email)) {
            Toast.makeText(this, "Akses ditolak. Hanya admin yang dapat masuk.", Toast.LENGTH_LONG).show();
            return;
        }
        
        showLoading(true);
        
        auth.signInWithEmailAndPassword(email, password)
                .addOnCompleteListener(this, task -> {
                    showLoading(false);
                    
                    if (task.isSuccessful()) {
                        FirebaseUser user = auth.getCurrentUser();
                        if (user != null && isAdminUser(user.getEmail())) {
                            Toast.makeText(this, "Login berhasil", Toast.LENGTH_SHORT).show();
                            navigateToMainActivity();
                        } else {
                            auth.signOut();
                            Toast.makeText(this, "Akses ditolak", Toast.LENGTH_SHORT).show();
                        }
                    } else {
                        String errorMessage = task.getException() != null ? 
                                task.getException().getMessage() : "Login gagal";
                        Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show();
                    }
                });
    }
    
    private boolean isAdminUser(String email) {
        // Dalam implementasi nyata, bisa menggunakan custom claims atau role-based auth
        return email != null && (email.equals(ADMIN_EMAIL) || email.contains("admin"));
    }
    
    private void showLoading(boolean show) {
        progressBar.setVisibility(show ? View.VISIBLE : View.GONE);
        btnLogin.setEnabled(!show);
    }
    
    private void navigateToMainActivity() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }
}

File: Tugas Video/PROJECT ANDROID STUDIO (PLAYLIST 2)/Wave Of Food User Dan Admin/WaveOfFood/app/src/main/java/com/kelasxi/waveoffood/LoginActivity.kt
package com.kelasxi.waveoffood

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

/**
 * Activity untuk halaman login pengguna
 */
class LoginActivity : AppCompatActivity() {
    
    private lateinit var etEmail: EditText
    private lateinit var etPassword: EditText
    private lateinit var btnLogin: Button
    private lateinit var tvRegister: TextView
    private lateinit var auth: FirebaseAuth
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        
        // Inisialisasi views
        etEmail = findViewById(R.id.etEmail)
        etPassword = findViewById(R.id.etPassword)
        btnLogin = findViewById(R.id.btnLogin)
        tvRegister = findViewById(R.id.tvRegister)
        
        // Inisialisasi Firebase Auth
        auth = FirebaseAuth.getInstance()
        
        setupClickListeners()
    }
    
    /**
     * Setup click listeners untuk button dan text view
     */
    private fun setupClickListeners() {
        // OnClickListener untuk tombol Login
        btnLogin.setOnClickListener {
            performLogin()
        }
        
        // OnClickListener untuk TextView Register
        tvRegister.setOnClickListener {
            startActivity(Intent(this, RegisterActivity::class.java))
        }
    }
    
    /**
     * Proses login pengguna
     */
    private fun performLogin() {
        val email = etEmail.text.toString().trim()
        val password = etPassword.text.toString().trim()
        
        // Validasi input
        if (email.isEmpty()) {
            etEmail.error = "Email tidak boleh kosong"
            etEmail.requestFocus()
            return
        }
        
        if (password.isEmpty()) {
            etPassword.error = "Password tidak boleh kosong"
            etPassword.requestFocus()
            return
        }
        
        // Disable tombol login saat proses
        btnLogin.isEnabled = false
        btnLogin.text = "Loading..."
        
        // Proses login menggunakan Firebase Auth
        auth.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(this) { task ->
                // Enable kembali tombol login
                btnLogin.isEnabled = true
                btnLogin.text = "Login"
                
                if (task.isSuccessful) {
                    // Login berhasil
                    Toast.makeText(this, "Login Berhasil", Toast.LENGTH_SHORT).show()
                    
                    // Arahkan ke MainActivity
                    startActivity(Intent(this, MainActivity::class.java))
                    finish() // Tutup LoginActivity
                } else {
                    // Login gagal
                    val errorMessage = task.exception?.message ?: "Login gagal"
                    Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()
                }
            }
    }
}
